### PR TITLE
[SDK-4328] Add support for all Client Addons listed in Management API

### PR DESCRIPTION
**Pull request by ewanharris**
### 🔧 Changes

Adds back support for all of the Client Addons listed in the [Management UI](https://auth0.com/docs/api/management/v2#!/Clients/get_clients_by_id), empty structs exist because the existence of the property is used to change behaviour

The structs were generated from the API directly.

### 📚 References

https://auth0.com/docs/api/management/v2#!/Clients/get_clients_by_id

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [ ] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: management/client.go [hunk 1]
```diff
@@ -232,14 +232,160 @@ type OIDCBackchannelLogout struct {
...
+	SAML2                *SAML2ClientAddon                `json:"samlp,omitempty"`
+	SAPAPI               *SAPAPIClientAddon               `json:"sap_api,omitempty"`
+	Sharepoint           *SharepointClientAddon           `json:"sharepoint,omitempty"`
+	Sentry               *SentryClientAddon               `json:"sentry,omitempty"`
+	Slack                *SlackClientAddon                `json:"slack,omitempty"`
+	SpringCM             *SpringCMClientAddon             `json:"springcm,omitempty"`
+	SSOIntegration       *SSOntegrationClientAddon        `json:"sso_integration,omitempty"`
+	WAMS                 *WAMSClientAddon                 `json:"wams,omitempty"`
 	// SAML2 Addon configuration. Set this property to indicate that the Addon should be enabled, all settings are optional.
 	// The first entry in `Callbacks` should be the URL to post the SAML Token to.
```
Comment (Widcket) [inline management/client.go:262]: Should this comment be moved above SAML2?

File: management/client.go [hunk 2]
```diff
@@ -303,6 +449,59 @@ type SAML2ClientAddonLogout struct {
...
+}
+
+// SlackClientAddon defines the `slack` settings for a client.
+type SlackClientAddon struct {
+	Team *string `json:"team,omitempty"`
+}
+
+// SpringCMClientAddon defines the `springcm` settings for a client.
+type SpringCMClientAddon struct {
+	Acsurl *string `json:"acsurl,omitempty"`
```
Comment (Widcket) [inline management/client.go:481]: Should this property be `AcsURL` instead?

Comment (Widcket) [reply management/client.go:481]: Or `ACSURL`?

File: management/client.go [hunk 3]
```diff
@@ -232,14 +232,160 @@ type OIDCBackchannelLogout struct {
...
+	MSCRM                *MSCRMClientAddon                `json:"mscrm,omitempty"`
+	NewRelic             *NewRelicClientAddon             `json:"newrelic,omitempty"`
+	Office365            *Office365ClientAddon            `json:"office365,omitempty"`
+	RMS                  *RMSClientAddon                  `json:"rms,omitempty"`
+	Salesforce           *SalesforceClientAddon           `json:"salesforce,omitempty"`
+	SalesforceAPI        *SalesforceAPIClientAddon        `json:"salesforce_api,omitempty"`
+	SalesforceSandboxAPI *SalesforceSandboxAPIClientAddon `json:"salesforce_sandbox_api,omitempty"`
+	SAML2                *SAML2ClientAddon                `json:"samlp,omitempty"`
+	SAPAPI               *SAPAPIClientAddon               `json:"sap_api,omitempty"`
+	Sharepoint           *SharepointClientAddon           `json:"sharepoint,omitempty"`
```
Comment (Widcket) [inline management/client.go:255]: Should this one be `SharePoint`? Also `SharePointClientAddon`.

File: management/client.go [hunk 4]
```diff
@@ -303,6 +449,59 @@ type SAML2ClientAddonLogout struct {
...
+	Team *string `json:"team,omitempty"`
+}
+
+// SpringCMClientAddon defines the `springcm` settings for a client.
+type SpringCMClientAddon struct {
+	Acsurl *string `json:"acsurl,omitempty"`
+}
+
+// SSOntegrationClientAddon defines the `sso_integration` settings for a client.
+type SSOntegrationClientAddon struct {
```
Comment (Widcket) [inline management/client.go:485]: Should this one be `SSOIntegrationClientAddon`?

Comment (ewanharris) [reply management/client.go]: Gah yeah absolutely, I must have removed the I when fixing the casing 🤦🏻

File: management/client.go (hunk 3 above)
Comment (ewanharris) [reply management/client.go]: Yup! I knew I should googled that one to check the casing.

File: management/client.go (hunk 2 above)
Comment (ewanharris) [reply management/client.go]: Fixed to be `ACSURL`

File: management/client.go (hunk 1 above)
Comment (ewanharris) [reply management/client.go:254]: Comment moved

**Pull request merged**